pkg/reconciler/taskrun/resources/input_resources.go

```diff
@@ -75,7 +75,12 @@ func AddInputResource(
 	for i := len(taskSpec.Inputs.Resources) - 1; i >= 0; i-- {
 		input := taskSpec.Inputs.Resources[i]
 		boundResource, err := getBoundResource(input.Name, taskRun.Spec.Inputs.Resources)
-		if err != nil {
+		// Continue if the declared resource is optional and not specified in TaskRun
+		// boundResource is nil if the declared resource in Task does not have any resource specified in the TaskRun
+		if input.Optional && boundResource == nil {
+			continue
+			//	throw an error for required resources, if not specified in the TaskRun
```

Comment:
v minor nit: this comment doesnt line up that nicely with the `else if` below it. consider removing or making it part of the comment above the if block or putting it in the else if block.

(same comment for similar section in `output_resource.go`)

Reply:
Done